src/Decompiler/Analysis/SsaTransform.cs

```diff
@@ -1357,29 +1357,19 @@ private static void GeneratePhiFunction(SsaIdentifier phi, SsaIdentifier[] sids)
             /// <returns>true if the phi function was trivial</returns>
             private bool TryRemoveTrivial(SsaIdentifier phi, out SsaIdentifier sid)
             {
-                Identifier same = null;
-                var phiFunc = ((PhiAssignment)phi.DefStatement.Instruction).Src;
+                var phiFunc = ((PhiAssignment) phi.DefStatement.Instruction).Src;
                 DebugEx.PrintIf(trace.TraceVerbose, "  Checking {0} for triviality", phiFunc);
-                foreach (Identifier op in phiFunc.Arguments)
-                {
-                    if (op == same || op == phi.Identifier)
-                        continue;
-                    if (same != null)
-                    {
-                        sid = null;
-                        return false;
-                    }
-                    same = op;
-                }
-                if (same == null)
+                if (!phiFunc.Arguments.Any(a => a != phi.Identifier))
```

Comment:
The double negative is confusing here, now. Isn't this the same as saying 
```suggestion
                if (phiFunc.Arguments.All(a => a == phi.Identifier))
```

Reply:
Changed. See 9f76399